### PR TITLE
For submissions and clarifications, prefix external ID in shadow mode

### PR DESCRIPTION
**Pull request by nickygerritsen**
This is needed to prevent collisions.

We can also use this logic to determine if someone submitted in DOMjudge while being shadow (i.e. an analyst instance) to make sure we judge the submission. This fixes the few breaking tests.

### CODE REVIEW

File: webapp/src/Entity/Clarification.php [hunk 1]
```diff
@@ -33,7 +33,8 @@
 #[UniqueEntity(fields: 'externalid')]
 class Clarification extends BaseApiEntity implements
     HasExternalIdInterface,
-    ExternalIdFromInternalIdInterface
+    ExternalIdFromInternalIdInterface,
+    PrefixedExternalIdInShadowModeInterface
```
Comment (vmcj) [inline webapp/src/Entity/Clarification.php:37]: Maybe document for which case we get this locally? We always expect to read clarifications from the eventfeed but don't disallow clarifications to be sent from the local instance but are never expected to be used outside of the system. For clarifications we should even not put them in the eventfeed if they are local in case we need to swap systems in the future?

File: webapp/src/Entity/Submission.php [hunk 2]
```diff
@@ -38,7 +38,8 @@
 #[UniqueEntity(fields: 'externalid')]
 class Submission extends BaseApiEntity implements
     HasExternalIdInterface,
-    ExternalIdFromInternalIdInterface
+    ExternalIdFromInternalIdInterface,
+    PrefixedExternalIdInShadowModeInterface
```
Comment (vmcj) [inline webapp/src/Entity/Submission.php:42]: As stated above, this case can happen if people resubmit after changing the code, useful for both jury as analysts.

File: webapp/tests/Unit/Controller/Jury/QueueTaskControllerTest.php [hunk 3]
```diff
@@ -144,7 +144,7 @@ public function testLazy(bool $shadowMode, int $globalLazy, int $problemLazy): v
         $config   = self::getContainer()->get(ConfigurationService::class);
         $eventLog = self::getContainer()->get(EventLogService::class);
         $dj       = self::getContainer()->get(DOMJudgeService::class);
-        $config->saveChanges(['lazy_eval_results'=>$globalLazy], $eventLog, $dj);
+        $config->saveChanges(['lazy_eval_results'=>$globalLazy], $eventLog, $dj, treatMissingBooleansAsFalse: false);
```
Comment (vmcj) [inline webapp/tests/Unit/Controller/Jury/QueueTaskControllerTest.php:147]: I don't know why this change is needed?

**Review by vmcj — APPROVED** [review]

File: webapp/tests/Unit/Controller/Jury/QueueTaskControllerTest.php (hunk 3 above)
Comment (nickygerritsen) [reply webapp/tests/Unit/Controller/Jury/QueueTaskControllerTest.php:147]: Since we don't resupply the boolean shadow mode, it would assume it is false again without this option

File: webapp/src/Entity/Clarification.php (hunk 1 above)
Comment (nickygerritsen) [reply webapp/src/Entity/Clarification.php:37]: Added a comment to the interface itself.

**Review by vmcj — APPROVED** [review]
